### PR TITLE
Ref #CDRW-685  Step 2 of the OAuth Hybrid Flow

### PR DESCRIPTION
**Pull request by splig**
SEE ALSO https://github.com/OpenBankingUK/reference-mock-server/pull/20 
- Refactor Session handler to store more than just session ID
     - Skeleton Code for Step 2 of the OAuth Hybrid Flow for Account Info Request
     - Actually calling out to the Read Write Mock Server ( still to be modified - see separate pull request )
     - Fixes for Request Promise Native
     - NEARLY working
     - Update Package Lock
     - Disabling Middleware for Account Request Id Request temporarily ...

### CODE REVIEW

File: .env.sample [hunk 1]
```diff
@@ -1,15 +1,21 @@
 PORT=8003
 DEBUG=error,log
 ASPSP_READWRITE_HOST=localhost:8001
+ASPSP_AUTH_SERVER=http://localhost:8001
 OB_PROVISIONED=false
+ASPSP_READWRITE_HOST=http://localhost:8001
```
Comment (robmckinnon) [inline .env.sample]: Two `ASPSP_READWRITE_HOST` in `.env.sample` - let's remove the unneeded one.

File: .env.sample [hunk 2]
```diff
@@ -1,15 +1,21 @@
...
+ASPSP_AUTH_ENDPOINT=http://localhost:8001/token
```
Comment (robmckinnon) [inline .env.sample]: Do we need both `ASPSP_AUTH_ENDPOINT` and `ASPSP_AUTH_SERVER`? They seem to express similar information.

File: test/ob-directory.test.js [hunk 3]
```diff
@@ -94,7 +94,7 @@ const login = application => request(application)
 describe('Directory', () => {
   session.setId('foo');
 
-  it('returns proxy 200 response for /account-payment-service-provider-authorisation-servers', (done) => {
+  xit('returns proxy 200 response for /account-payment-service-provider-authorisation-servers', (done) => {
```
Comment (robmckinnon) [inline test/ob-directory.test.js]: Anything preventing this test get re-enabled?

File: test/index.test.js [hunk 4]
```diff
@@ -132,10 +130,9 @@ describe('Session Deletion (Logout)', () => {
 });
 
 describe('Proxy', () => {
-  it('returns proxy 200 response for /open-banking/v1.1/accounts with valid session', (done) => {
+  xit('returns proxy 200 response for /open-banking/v1.1/accounts with valid session', (done) => {
```
Comment (robmckinnon) [inline test/index.test.js]: Anything preventing this test get re-enabled?

File: app/account-request.js [hunk 5]
```diff
@@ -0,0 +1,66 @@
...
+const requestPromise = require('request-promise-native');
+
+const loopbackHost = process.env.LOOPBACK_HOST || '';
+
+let acctRefId = false; // SPOOF instead of Persistence like mongo
+
+// Get the PSU from the session for storage of WHO owns the AccontRequestId
+const getPsuIdFromSid = (sid, cb) => {
+  log(` retrieving PSU ID from SID ${sid}`);
+  return cb('alice');
```
Comment (robmckinnon) [inline app/account-request.js]: Maybe not use callbacks? Prefer we make these `async` functions. Let's discuss. :)

File: app/flows/hybrid.js [hunk 6]
```diff
@@ -0,0 +1,217 @@
...
+const Step1PsuReqAccountInfo = (() => {
+  /**
+   * This is the code that takes the Account Information Request
+   * from the Client and then kicks off the Account Request
+   */
+  const getAccountInfo = (a, b) => new Promise((resolve, reject) => {
+    if (a === b) { // TODO - proper implemetation!
+      resolve('Success');
+    } else {
+      reject(new Error('Something went wrong'));
```
Comment (robmckinnon) [inline app/flows/hybrid.js]: Maybe switch to use `async` syntax instead of `new Promise`? Let's discuss. :)

File: app/flows/hybrid.js [hunk 7]
```diff
@@ -0,0 +1,217 @@
...
+    }
+  });
+
+  return {
+    getAccountInfo,
+  };
+})();
+
+const Step2SetupAccountRequest = (() => {
```
Comment (daniel-kuras) [inline app/flows/hybrid.js]: Can we break the current design to functional components instead of steps based components?

**Comment by splig** [comment]
This PR no longer needed

**Pull request closed without merging**